### PR TITLE
Ignore blank import of embed if embed is actually used in the file

### PR DESCRIPTION
**Pull request by grongor**
Closes https://github.com/mgechev/revive/issues/500

### CODE REVIEW

File: rule/blank-imports.go [hunk 1]
```diff
@@ -52,14 +55,20 @@ func (w lintBlankImports) Visit(_ ast.Node) ast.Visitor {
...
 			prev := w.fileAst.Imports[i-1]
 			prevPos := w.file.ToPosition(prev.Pos())
-			if isBlank(prev.Name) && prevPos.Line+1 == pos.Line {
+
+			if isBlank(prev.Name) && prevPos.Line+1 == pos.Line && prev.Path.Value != embedImportPath {
 				continue // A subsequent blank in a group.
 			}
 		}
 
+		if imp.Path.Value == embedImportPath && w.fileHasValidEmbedComment() {
```
Comment (chavacava) [inline rule/blank-imports.go:68]: we do not need to call `w.fileHasValidEmbedComment()` inside the loop. In fact we can call it just once at the beginning of the analysis. 
If we do this, and the file effectively has a valid embed comment, do we need to check the imports?

File: rule/blank-imports.go [hunk 2]
```diff
@@ -52,14 +55,20 @@ func (w lintBlankImports) Visit(_ ast.Node) ast.Visitor {
 		if !isBlank(imp.Name) {
 			continue // Ignore non-blank imports.
 		}
+
 		if i > 0 {
 			prev := w.fileAst.Imports[i-1]
 			prevPos := w.file.ToPosition(prev.Pos())
-			if isBlank(prev.Name) && prevPos.Line+1 == pos.Line {
+
+			if isBlank(prev.Name) && prevPos.Line+1 == pos.Line && prev.Path.Value != embedImportPath {
```
Comment (chavacava) [inline rule/blank-imports.go:63]: we could add a variable to document the semantics of this condition. Something like:
 ```suggestion
			isSubsequentBlancInAGroup := isBlank(prev.Name) && prevPos.Line+1 == pos.Line && prev.Path.Value != embedImportPath 
			if  isSubsequentBlancInAGroup {
```

**Review by chavacava — CHANGES_REQUESTED** [review]
Hi @grongor, thanks for the PR. I've made a couple of remarks.

File: rule/blank-imports.go (hunk 2 above)
Comment (grongor) [reply rule/blank-imports.go:63]: Sure, why not :)

File: rule/blank-imports.go (hunk 1 above)
Comment (grongor) [reply rule/blank-imports.go:69]: Well, the embed import should be present in the file only once, so this method is effectively called only once. And for files that do not use embed (which will be majority of files) we don't do unnecessary comment analysis. From performance point of view I think my approach is better. But this is my first time working with Go AST so I might be wrong...

Comment (chavacava) [reply rule/blank-imports.go:69]: @grongor you ar right, my bad. The rule logic is not based on AST traversal and can be directly implemented in the `Apply` method without requiring to use a tree walker object.

Comment (grongor) [reply rule/blank-imports.go]: I think it's out of the scope for this MR/issue, but sure, it makes sense to not use the `ast.Walk` here ... I added the commit that fixes that.

**Review by chavacava — APPROVED** [review]

**Comment by chavacava** [comment]
@grongor thanks for the PR and sorry for the late response (it was a busy week)

**Comment by grongor** [comment]
No worries, thanks for merging :)

**Pull request merged**